### PR TITLE
#654 Letter Avatar component implementation

### PR DESCRIPTION
**Pull request by aripddev**

### CODE REVIEW

File: src/main/java/org/primefaces/extensions/component/letteravatar/LetteravatarRenderer.java [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+import java.io.IOException;
+import javax.faces.component.UIComponent;
+import javax.faces.context.FacesContext;
+import javax.faces.context.ResponseWriter;
+import javax.faces.render.FacesRenderer;
+import javax.faces.render.Renderer;
+
+@FacesRenderer(componentFamily = Letteravatar.COMPONENT_FAMILY, rendererType = Letteravatar.DEFAULT_RENDERER)
+public class LetteravatarRenderer extends Renderer {
+
```
Comment (melloware) [inline src/main/java/org/primefaces/extensions/component/letteravatar/LetteravatarRenderer.java]: Lets capitalize the A so LetterAvatarRenderer.

File: src/main/java/org/primefaces/extensions/component/letteravatar/Letteravatar.java [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+    @ResourceDependency(library = "letteravatar", name = "letteravatar.js")})
+public class Letteravatar extends UIComponentBase {
+
+    public static final String COMPONENT_TYPE = "org.primefaces.extensions.component.Letteravatar";
+    public static final String COMPONENT_FAMILY = "org.primefaces.extensions.component";
+    public static final String DEFAULT_RENDERER = "org.primefaces.extensions.component.LetteravatarRenderer";
+
+    protected static enum PropertyKeys {
+        widgetVar,
+        style, styleClass,
```
Comment (melloware) [inline src/main/java/org/primefaces/extensions/component/letteravatar/Letteravatar.java]: Remove widgetVar as it looks like this component has no client side widget.

File: src/main/java/org/primefaces/extensions/component/letteravatar/LetteravatarRenderer.java [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+        final Integer height = letteravatar.getHeight();
+        final String avatar = letteravatar.getValue();
+        final Boolean rounded = letteravatar.isRounded();
+
+        writer.writeAttribute("width", width, null);
+        writer.writeAttribute("height", height, null);
+        writer.writeAttribute("avatar", avatar, null);
+        if (rounded) {
+            writer.writeAttribute("style", "border-radius: 50%;", null);
+        }
```
Comment (melloware) [inline src/main/java/org/primefaces/extensions/component/letteravatar/LetteravatarRenderer.java]: I think if you do this rounded you will override their style they added above.  Look at some of hte other components they add the style together before add the style attribute so you can append their style to this rounded one.

**Review by melloware — CHANGES_REQUESTED** [review]

**Comment by aripddev** [comment]
I have completed the changes. Do I need to create a new PR for the corrections?

**Comment by melloware** [comment]
Nope as long as you made the commits on this branch i pick them up!  I will review and merge tomorrow!

**Pull request merged**